flax/linen/module.py

```diff
@@ -491,49 +608,103 @@ def param(self, name: str, init_fn: Callable[..., T], *init_args) -> T:
     self.children[name] = 'params'
     return v
 
-  def get_variable(self, kind: str, name: str, default: T = None) -> T:
-    """Get raw value of variable on this Module."""
-    return self.scope.get_variable(kind, name, default)
+  def has_variable(self, col: str, name: str) -> bool:
+    """Checks if a variable of given collection and name exists in this Module.
+
+    See :mod:`flax.core.variables` for more explanation on variables and 
+    collections.
+    
+    Args:
+      col: The variable collection name.
+      name: The name of the variable.
+    Returns:
+      True if the variable exists.
+    """
+    return self.scope.has_variable(col, name)
 
-  def has_variable(self, kind: str, name: str):
-    """Check if a variable of given kind and name exists in this Module."""
-    return self.scope.has_variable(kind, name)
+  def make_rng(self, col: str) -> PRNGKey:
+    """Returns a new RNG key for a given collection for this Module. 
+    
+    The new RNG key is split from the previous one. Thus, every call to 
+    `make_rng` returns a new RNG key, while still guaranteeing full
+    reproducibility.
 
-  def put_variable(self, kind: str, name: str, value: Any):
-    """Direectly set value of a variable on this Module."""
-    return self.scope.put_variable(kind, name, value)
+    TODO: Link to both Flax RNG design note.
 
-  def make_rng(self, kind: str) -> PRNGKey:
-    """Get a new rng key of a given kind from this Module."""
-    return self.scope.make_rng(kind)
+    Args:
+      col: The variable collection name.
+    Returns:
+      The newly generated RNG key.
+    """
+    return self.scope.make_rng(col)
 
-  def apply(self, variables, *args, rngs=None,
-            method=None, mutable=False, **kwargs):
-    """Apply module to variables and return output and modified variables."""
+  def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
+            method: Callable[..., Any] = None, 
+            mutable: Union[bool, str, Sequence[str]] = False,
+            **kwargs) -> Union[Any, Tuple[Any, VariableDict]]:
+    """Applies a module method to variables and returns output and modified variables.
+    
+    Args:
+      variables: A dictionary containing variables keyed by variable 
+        collections. See :mod:`flax.core.variables` for more details
+        about variables.
+      rngs: The rngs for the variable collections.
+      method: An optional method. If provided, applies this method. If not
```

Comment:
we should clarify here and elsewhere that you literally pass the method i.e. `mymodule.mymethod` which people find weird as hell and may change, but we should at least be clear about it for now.

Reply:
I agree. I've updated the comment.